readme.md

```diff
@@ -17,39 +17,81 @@ $ npm install alpha-sort
 ```js
 const alphaSort = require('alpha-sort');
 
-['b', 'a', 'c'].sort(alphaSort.ascending);
+['b', 'a', 'c'].sort(alphaSort());
 //=> ['a', 'b', 'c']
+
+['b', 'a', 'c'].sort(alphaSort({descending: true}));
+//=> ['c', 'b', 'a']
+
+['B', 'a', 'C'].sort(alphaSort({caseInsensitive: true}));
+//=> ['a', 'B', 'C']
+
+['file10.txt', 'file2.txt', 'file03.txt'].sort(alphaSort({natural: true}));
+//=> ['file2.txt', 'file03.txt', 'file10.txt']
 ```
 
 
 ## API
 
-### alphaSort.ascending
+### alphaSort(options?)
+
+Get a comparator function to be used as argument for [`Array#sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).
+
+#### options
 
-Ascending sort comparator.
+Type: `object`
 
-### alphaSort.descending
+##### descending
 
-Descending sort comparator.
+Type: `boolean`\
+Default: `false`
 
-### alphaSort.caseInsensitiveAscending
+Whether or not to sort descending.
 
-Case-insensitive ascending sort comparator.
+##### caseInsensitive
 
-Note: If two elements are considered equal in the case-insensitive comparison, the tie-break will be a case-sensitive comparison:
+Type: `boolean`\
+Default: `false`
+
+Whether or not to sort in a case-insensitive way.
+
+Note: If two elements are considered equal in the case-insensitive comparison, the tie-break will be a standard (case-sensitive) comparison. Example:
 
 ```js
 const alphaSort = require('alpha-sort');
 
-['bar', 'baz', 'Baz'].sort(alphaSort.caseInsensitiveAscending);
+['bar', 'baz', 'Baz'].sort(alphaSort({caseInsensitive: true}));
 //=> ['bar', 'Baz', 'baz']
 ```
 
-### alphaSort.caseInsensitiveDescending
+##### natural
+
+Type: `boolean`\
+Default: `false`
+
+Whether or not to sort using [natural sort order](https://en.wikipedia.org/wiki/Natural_sort_order).
 
-Case-insensitive descending sort comparator.
+Note: If two elements are considered equal in the natural sort order comparison, the tie-break will be a standard (non-natural) comparison.
+
+##### preprocessor
+
+Type: `function`\
+Default: `undefined`
+
+A custom function that you can provide to manipulate the elements before sorting. This does not modify the values of the array; it only interferes in the sorting order.
+
+This can be used, for example, if you are sorting book titles in English and want to ignore common articles such as `the`, `a` or `an`:
+
+```js
+const alphaSort = require('alpha-sort');
+
+['The Foo', 'Bar'].sort(alphaSort({
+	preprocessor: title => title.replace(/^(the|a|an) /i, '')
```

Comment:
What do you think of bundling this preprocessor for convenience?

```suggestion
	preprocessor: title => title.replace(/^(?:the|a|an) /i, '')
```

Reply:
Hello, I've [applied](https://github.com/sindresorhus/alpha-sort/commit/8b307eb5b936622cce9b84776fe6dba6846ee959#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R89) the suggested change.

> What do you think of bundling this preprocessor for convenience?

Personally I don't like the idea. I think this use case is too specific. I'd rather keep it as just an example, if someone needs this, they can just copy this code, it is just one line. Also, I can see people wanting to use `\s+` in the end of that regex, instead of a single space. Or also accepting `-` as an article separator, instead of only considering a space. Or `_`. Therefore I think whichever choice we make, for the rare cases where someone would like this, we might not be providing the perfect preprocessor anyway.

Reply:
Alright. Agreed.